### PR TITLE
feat(NumberInput): add experimental NumberInput

### LINKED ISSUE #1710: Form validation UI content text should not influence styling of number input up/down buttons
## When utilizing the number input, but stripping out the helper text/label/Form validation UI content, but clicking on the form validation, the number buttons are out of place int he input field

> See picture

> Is this issue related to a specific component? number input

> What did you expect to happen? What happened instead? What would you like to see changed?

I'd expect given the content the number input buttons not to have it's styling influenced

> What browser are you working in? Chrome Version 71.0.3578.98 (Official Build) (64-bit)

> What version of the Carbon Design System are you using?     carbon-components-react": "^6.58.0",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
Yes
## Steps to reproduce the issue
See comments/screen capture
image

### PR DESCRIPTION
**Pull request by emyarod**
Closes IBM/carbon-components-react#1653
Closes #1710 
Closes https://github.com/carbon-design-system/carbon-website/issues/827

This PR will arrange the NumberInput to match the vanilla spec. The `form-item` div wrapper is removed as we have an explicit `<FormItem>` component in the library that already accomplishes this.

This PR also ties `no-label` classes to the value of the `label` prop in experimental mode rather than the explicit `hideLabel` prop. Looking for feedback on this change which would eventually be reflected across the library as well.

#### Changelog

**Changed**

- use updated markup
- tie `no-label` classes to the value of `label` rather than explicit `hideLabel` prop

**Removed**

- unit test searching for `<FormItem>` class

### CODE REVIEW

**Comment by asudoh** [comment]
Thanks @emyarod for this proposal! As I DM'ed, great to see `hideLabel` removal is done across the board and probably de-coupled from this PR.

**Comment by emyarod** [comment]
issue opened @asudoh https://github.com/IBM/carbon-components-react/issues/1662

I will begin development in a separate branch

**Comment by joshblack** [comment]
Do we need to guard these changes behind a feature flag at all? Was going through it quickly and that question came to mind given the changes, just wanted to ask!

File: src/components/NumberInput/NumberInput.js [hunk 1]
```diff
@@ -252,43 +252,41 @@ export default class NumberInput extends Component {
     ) : null;
 
     return (
-      <div className={`${prefix}--form-item`}>
```
[the remark this answers is not in the document]
Comment (emyarod) [reply src/components/NumberInput/NumberInput.js]: yeah I think I can avoid the breaking change for now, just spotted inconsistency in using `<FormItem />` across components and storybook examples. I can open this discussion in a separate issue if that would be better

[the remark this answers is not in the document]
Comment (emyarod) [reply src/components/NumberInput/NumberInput.js]: created an issue to track this here https://github.com/IBM/carbon-components-react/issues/1681

File: src/components/NumberInput/NumberInput.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ export default class NumberInput extends Component {
     disabled: false,
     hideLabel: false,
     iconDescription: 'choose a number',
-    label: ' ',
+    label: '',
```
Comment (asudoh) [inline src/components/NumberInput/NumberInput.js]: This looks a breaking change, do we want this change for experimental mode only...?

Comment (asudoh) [reply src/components/NumberInput/NumberInput.js]: @emyarod Thoughts...? Thanks!

Comment (emyarod) [reply src/components/NumberInput/NumberInput.js]: updated!

**Comment by emyarod** [comment]
possibly dependent on https://github.com/IBM/carbon-components/pull/1604

**Comment by emyarod** [comment]
depends on https://github.com/IBM/carbon-components/issues/1621

**Comment by emyarod** [comment]
PR has been updated to reflect changes from https://github.com/IBM/carbon-components/pull/1622

**Comment by emyarod** [comment]
I thought about that @asudoh but wasn't sure if it was easier to read. I went ahead with the change though

**Comment by asudoh** [comment]
Seems that all comments have been addressed, great to have another review so we can go forward with this. Thanks!

**Comment by asudoh** [comment]
Thank you for reviewing @lovemecomputer!

**Pull request merged**